admin/server/auth/handlers.go

```diff
@@ -24,17 +27,25 @@ const (
-func (a *Authenticator) RegisterEndpoints(mux *http.ServeMux) {
+func (a *Authenticator) RegisterEndpoints(mux *http.ServeMux, limiter ratelimit.Limiter) {
 	// TODO: Add helper utils to clean this up
 	inner := http.NewServeMux()
-	inner.Handle("/auth/login", otelhttp.WithRouteTag("/auth/login", http.HandlerFunc(a.authLogin)))
-	inner.Handle("/auth/callback", otelhttp.WithRouteTag("/auth/callback", http.HandlerFunc(a.authLoginCallback)))
-	inner.Handle("/auth/with-token", otelhttp.WithRouteTag("/auth/with-token", http.HandlerFunc(a.authWithToken)))
-	inner.Handle("/auth/logout", otelhttp.WithRouteTag("/auth/logout", http.HandlerFunc(a.authLogout)))
-	inner.Handle("/auth/logout/callback", otelhttp.WithRouteTag("/auth/logout/callback", http.HandlerFunc(a.authLogoutCallback)))
-	inner.Handle("/auth/oauth/device_authorization", otelhttp.WithRouteTag("/auth/oauth/device_authorization", http.HandlerFunc(a.handleDeviceCodeRequest)))
-	inner.Handle("/auth/oauth/device", otelhttp.WithRouteTag("/auth/oauth/device", a.HTTPMiddleware(http.HandlerFunc(a.handleUserCodeConfirmation)))) // NOTE: Uses auth middleware
-	inner.Handle("/auth/oauth/token", otelhttp.WithRouteTag("/auth/oauth/token", http.HandlerFunc(a.getAccessToken)))
+	inner.Handle("/auth/login", otelhttp.WithRouteTag("/auth/login",
+		LimiterHTTPHandler("/auth/login", limiter, ratelimit.Sensitive, http.HandlerFunc(a.authLogin))))
+	inner.Handle("/auth/callback", otelhttp.WithRouteTag("/auth/callback",
+		LimiterHTTPHandler("/auth/callback", limiter, ratelimit.Sensitive, http.HandlerFunc(a.authLoginCallback))))
+	inner.Handle("/auth/with-token", otelhttp.WithRouteTag("/auth/with-token",
+		LimiterHTTPHandler("/auth/with-token", limiter, ratelimit.Sensitive, http.HandlerFunc(a.authWithToken))))
+	inner.Handle("/auth/logout", otelhttp.WithRouteTag("/auth/logout",
+		LimiterHTTPHandler("/auth/logout", limiter, ratelimit.Sensitive, http.HandlerFunc(a.authLogout))))
+	inner.Handle("/auth/logout/callback", otelhttp.WithRouteTag("/auth/logout/callback",
+		LimiterHTTPHandler("/auth/logout/callback", limiter, ratelimit.Sensitive, http.HandlerFunc(a.authLogoutCallback))))
+	inner.Handle("/auth/oauth/device_authorization", otelhttp.WithRouteTag("/auth/oauth/device_authorization",
+		LimiterHTTPHandler("/auth/oauth/device_authorization", limiter, ratelimit.Sensitive, http.HandlerFunc(a.handleDeviceCodeRequest))))
+	inner.Handle("/auth/oauth/device", otelhttp.WithRouteTag("/auth/oauth/device",
+		LimiterHTTPHandler("/auth/oauth/device", limiter, ratelimit.Sensitive, a.HTTPMiddleware(http.HandlerFunc(a.handleUserCodeConfirmation))))) // NOTE: Uses auth middleware
+	inner.Handle("/auth/oauth/token", otelhttp.WithRouteTag("/auth/oauth/token",
+		LimiterHTTPHandler("/auth/oauth/token", limiter, ratelimit.Sensitive, http.HandlerFunc(a.getAccessToken))))
 	mux.Handle("/auth/", observability.Middleware("admin", a.logger, inner))
```

Comment:
This sets a single key for all handlers. Do we want it to be one for all handlers or on per route basis?

Reply:
If it is set for `/auth/` only then it is possible to look at `req.Path` in order to customize a limit or make an exclusion but in general this isn't right because path and route might be not equal